tiled/catalog/adapter.py

```diff
@@ -1375,6 +1388,37 @@ def specs(query, tree):
     return tree.new_variation(conditions=tree.conditions + conditions)
 
 
+def access_blob_filter(query, tree):
+    dialect_name = tree.engine.url.get_dialect().name
+    access_blob = orm.Node.access_blob
+    if len(query.user_id) == 0 and len(query.tags) == 0:
```

Comment:
This caught me by surprise because I would expect `"user"` to be `None` (or in JSON parlance `null`) rather than `""`. This would catch either case:

```suggestion
    if (not query.user_id) and len(query.tags) == 0:
```

or even

```suggestion
    if not (query.user_id or query.tags)
```

as both empty strings and empty lists are false-y. YMMV on how that reads.

Reply:
Agreed, this was me just picking up the syntax from the other filters - I think I misinterpreted that a `len` check had been done on a string elsewhere.